### PR TITLE
Extending tests for sentieon joint germline AND avoiding duplicated variants in VQSR vcfs

### PR DESCRIPTION
**Pull request by asp8200**
Extending some CI-test for sentieon-based joint-germline. Inspired by what was done in PR #1173 

Concerning VQSR in joint-germline subworkflows:

The following GATK-based joint-germline cmd results in the joint-germeline-vcf-file being sent through SNP and INDEL recalibration (VARIANTRECALIBRATOR and VQSR): 
```
nextflow run main.nf -profile test_cache,tools_germline,docker --input ./tests/csv/3.0/mapped_joint_bam.csv --tools haplotypecaller --step variant_calling --joint_germline --outdir results2  -stub-run
```
Similarly for the Sentieon-based joint-germline cmd:
```
nextflow run main.nf -profile test_cache,tools_germline,docker --input ./tests/csv/3.0/mapped_joint_bam.csv --tools sentieon_haplotyper --step variant_calling --joint_germline --outdir results --sentieon_haplotyper_emit_mode gvcf -stub-run
```
It results in the  joint-germeline-vcf-file being sent through SNP and INDEL recalibration (SENTIEON_VARCAL and SENTIEON_APPLYVARCAL): 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/sentieon/applyvarcal/main.nf [hunk 1]
```diff
@@ -59,4 +59,20 @@ process SENTIEON_APPLYVARCAL {
...
+    }
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.vcf.gz
+    touch ${prefix}.vcf.gz.tbi
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        sentieon: \$(echo \$(sentieon driver --version 2>&1) | sed -e "s/sentieon-genomics-//g")
+    END_VERSIONS
```
Comment (maxulysse) [inline modules/nf-core/sentieon/applyvarcal/main.nf:76]: this should be done in modules

Comment (asp8200) [reply modules/nf-core/sentieon/applyvarcal/main.nf:76]: I know. I already wrote that on Slack. 😁  I wanna get this test passing first though

Comment (maxulysse) [reply modules/nf-core/sentieon/applyvarcal/main.nf:76]: sorry was looking only at github

File: subworkflows/local/bam_variant_calling_germline_all/main.nf [hunk 2]
```diff
@@ -223,11 +217,29 @@ workflow BAM_VARIANT_CALLING_GERMLINE_ALL {
                 known_sites_snps_tbi,
                 known_snps_vqsr)
 
+            // TO-DO: Should this VCF from the sentieon haploTYPER really be put into a variable called vcf_haploTYPECALLER
```
Comment (asp8200) [inline subworkflows/local/bam_variant_calling_germline_all/main.nf:220]: @maxulysse @FriederikeHanssen : Any thoughts on that?  I think it should be renamed to `vcf_haplotyper` and then that should be added to the annotation over in `sarek.nf`.

Comment (FriederikeHanssen) [reply subworkflows/local/bam_variant_calling_germline_all/main.nf:220]: yeah maybe keeping them separate is a good idea. That way you could run both if you really wanted to

Comment (asp8200) [reply subworkflows/local/bam_variant_calling_germline_all/main.nf:220]: I'll replace "my" usage of `vcf_haplotyper`with `vcf_sentieon_haplotyper`. This also seems to solve the problem of the unfiltered vcf ending up in the csv.

**Comment by maxulysse** [comment]
Looking good.
Can you update CHANGELOG?

**Review by maxulysse — APPROVED** [review]
LGTM

**Pull request merged**